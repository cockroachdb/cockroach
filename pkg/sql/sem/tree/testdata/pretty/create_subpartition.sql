CREATE TABLE students (
    id SERIAL,
    name STRING,
    email STRING,
    country STRING,
    expected_graduation_date DATE,
    PRIMARY KEY (country, expected_graduation_date, id))
    PARTITION BY LIST (country)(
        PARTITION australia VALUES IN ('AU','NZ') PARTITION BY RANGE (expected_graduation_date)(PARTITION graduated_au VALUES FROM (MINVALUE) TO ('2017-08-15'), PARTITION current_au VALUES FROM ('2017-08-15') TO (MAXVALUE)),
        PARTITION north_america VALUES IN ('US','CA') PARTITION BY RANGE (expected_graduation_date)(PARTITION graduated_us VALUES FROM (MINVALUE) TO ('2017-08-15'), PARTITION current_us VALUES FROM ('2017-08-15') TO (MAXVALUE))
    )
