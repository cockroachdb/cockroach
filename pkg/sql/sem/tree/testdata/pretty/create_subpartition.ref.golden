// Code generated by TestPretty. DO NOT EDIT.
// GENERATED FILE DO NOT EDIT
1:
-
CREATE TABLE students (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (
		country,
		expected_graduation_date,
		id
	)
)
	PARTITION BY LIST (
		country
	)
		(
			PARTITION australia
				VALUES IN (
					'AU',
					'NZ'
				)
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_au
							VALUES FROM (
								minvalue
							)
							TO (
								'2017-08-15'
							),
						PARTITION current_au
							VALUES FROM (
								'2017-08-15'
							)
							TO (
								maxvalue
							)
					),
			PARTITION north_america
				VALUES IN (
					'US',
					'CA'
				)
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_us
							VALUES FROM (
								minvalue
							)
							TO (
								'2017-08-15'
							),
						PARTITION current_us
							VALUES FROM (
								'2017-08-15'
							)
							TO (
								maxvalue
							)
					)
		)

31:
-------------------------------
CREATE TABLE students (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (
		country,
		expected_graduation_date,
		id
	)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN (
					'AU',
					'NZ'
				)
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_au
							VALUES FROM (
								minvalue
							)
							TO (
								'2017-08-15'
							),
						PARTITION current_au
							VALUES FROM (
								'2017-08-15'
							)
							TO (
								maxvalue
							)
					),
			PARTITION north_america
				VALUES IN (
					'US',
					'CA'
				)
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_us
							VALUES FROM (
								minvalue
							)
							TO (
								'2017-08-15'
							),
						PARTITION current_us
							VALUES FROM (
								'2017-08-15'
							)
							TO (
								maxvalue
							)
					)
		)

38:
--------------------------------------
CREATE TABLE students (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (
		country,
		expected_graduation_date,
		id
	)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_au
							VALUES FROM (
								minvalue
							)
							TO (
								'2017-08-15'
							),
						PARTITION current_au
							VALUES FROM (
								'2017-08-15'
							)
							TO (
								maxvalue
							)
					),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_us
							VALUES FROM (
								minvalue
							)
							TO (
								'2017-08-15'
							),
						PARTITION current_us
							VALUES FROM (
								'2017-08-15'
							)
							TO (
								maxvalue
							)
					)
		)

41:
-----------------------------------------
CREATE TABLE students (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (
		country,
		expected_graduation_date,
		id
	)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_au
							VALUES FROM (
								minvalue
							)
							TO (
								'2017-08-15'
							),
						PARTITION current_au
							VALUES FROM (
								'2017-08-15'
							)
							TO (maxvalue)
					),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_us
							VALUES FROM (
								minvalue
							)
							TO (
								'2017-08-15'
							),
						PARTITION current_us
							VALUES FROM (
								'2017-08-15'
							)
							TO (maxvalue)
					)
		)

46:
----------------------------------------------
CREATE TABLE students (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (
		country,
		expected_graduation_date,
		id
	)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_au
							VALUES FROM (
								minvalue
							)
							TO ('2017-08-15'),
						PARTITION current_au
							VALUES FROM (
								'2017-08-15'
							)
							TO (maxvalue)
					),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_us
							VALUES FROM (
								minvalue
							)
							TO ('2017-08-15'),
						PARTITION current_us
							VALUES FROM (
								'2017-08-15'
							)
							TO (maxvalue)
					)
		)

50:
--------------------------------------------------
CREATE TABLE students (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (
		country,
		expected_graduation_date,
		id
	)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_au
							VALUES FROM (minvalue)
							TO ('2017-08-15'),
						PARTITION current_au
							VALUES FROM (
								'2017-08-15'
							)
							TO (maxvalue)
					),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_us
							VALUES FROM (minvalue)
							TO ('2017-08-15'),
						PARTITION current_us
							VALUES FROM (
								'2017-08-15'
							)
							TO (maxvalue)
					)
		)

54:
------------------------------------------------------
CREATE TABLE students (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (
		country,
		expected_graduation_date,
		id
	)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_au
							VALUES FROM (minvalue)
							TO ('2017-08-15'),
						PARTITION current_au
							VALUES FROM ('2017-08-15')
							TO (maxvalue)
					),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_us
							VALUES FROM (minvalue)
							TO ('2017-08-15'),
						PARTITION current_us
							VALUES FROM ('2017-08-15')
							TO (maxvalue)
					)
		)

55:
-------------------------------------------------------
CREATE TABLE students (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (country, expected_graduation_date, id)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_au
							VALUES FROM (minvalue)
							TO ('2017-08-15'),
						PARTITION current_au
							VALUES FROM ('2017-08-15')
							TO (maxvalue)
					),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (
					expected_graduation_date
				)
					(
						PARTITION graduated_us
							VALUES FROM (minvalue)
							TO ('2017-08-15'),
						PARTITION current_us
							VALUES FROM ('2017-08-15')
							TO (maxvalue)
					)
		)

61:
-------------------------------------------------------------
CREATE TABLE students (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (country, expected_graduation_date, id)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (expected_graduation_date)
					(
						PARTITION graduated_au
							VALUES FROM (minvalue)
							TO ('2017-08-15'),
						PARTITION current_au
							VALUES FROM ('2017-08-15')
							TO (maxvalue)
					),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (expected_graduation_date)
					(
						PARTITION graduated_us
							VALUES FROM (minvalue)
							TO ('2017-08-15'),
						PARTITION current_us
							VALUES FROM ('2017-08-15')
							TO (maxvalue)
					)
		)

68:
--------------------------------------------------------------------
CREATE TABLE students (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (country, expected_graduation_date, id)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (expected_graduation_date)
					(
						PARTITION graduated_au
							VALUES FROM (minvalue)
							TO ('2017-08-15'),
						PARTITION current_au
							VALUES FROM ('2017-08-15') TO (maxvalue)
					),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (expected_graduation_date)
					(
						PARTITION graduated_us
							VALUES FROM (minvalue)
							TO ('2017-08-15'),
						PARTITION current_us
							VALUES FROM ('2017-08-15') TO (maxvalue)
					)
		)

69:
---------------------------------------------------------------------
CREATE TABLE students (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (country, expected_graduation_date, id)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (expected_graduation_date)
					(
						PARTITION graduated_au
							VALUES FROM (minvalue) TO ('2017-08-15'),
						PARTITION current_au
							VALUES FROM ('2017-08-15') TO (maxvalue)
					),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (expected_graduation_date)
					(
						PARTITION graduated_us
							VALUES FROM (minvalue) TO ('2017-08-15'),
						PARTITION current_us
							VALUES FROM ('2017-08-15') TO (maxvalue)
					)
		)

85:
-------------------------------------------------------------------------------------
CREATE TABLE students (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (country, expected_graduation_date, id)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (expected_graduation_date)
					(
						PARTITION graduated_au
							VALUES FROM (minvalue) TO ('2017-08-15'),
						PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)
					),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (expected_graduation_date)
					(
						PARTITION graduated_us
							VALUES FROM (minvalue) TO ('2017-08-15'),
						PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue)
					)
		)

88:
----------------------------------------------------------------------------------------
CREATE TABLE students (
	id
		SERIAL8,
	name
		STRING,
	email
		STRING,
	country
		STRING,
	expected_graduation_date
		DATE,
	PRIMARY KEY (country, expected_graduation_date, id)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (expected_graduation_date)
					(
						PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'),
						PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)
					),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (expected_graduation_date)
					(
						PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'),
						PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue)
					)
		)

89:
-----------------------------------------------------------------------------------------
CREATE TABLE students (
	id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE,
	PRIMARY KEY (country, expected_graduation_date, id)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (expected_graduation_date)
					(
						PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'),
						PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)
					),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (expected_graduation_date)
					(
						PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'),
						PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue)
					)
		)

148:
----------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students (
	id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE,
	PRIMARY KEY (country, expected_graduation_date, id)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (expected_graduation_date)
					(
						PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'),
						PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)
					),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (expected_graduation_date)
					(PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue))
		)

149:
-----------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students (
	id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE,
	PRIMARY KEY (country, expected_graduation_date, id)
)
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (expected_graduation_date)
					(PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (expected_graduation_date)
					(PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue))
		)

161:
-----------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, expected_graduation_date, id))
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (expected_graduation_date)
					(PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (expected_graduation_date)
					(PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue))
		)

190:
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, expected_graduation_date, id))
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (expected_graduation_date)
					(PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue))
		)

191:
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, expected_graduation_date, id))
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)),
			PARTITION north_america
				VALUES IN ('US', 'CA')
				PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue))
		)

213:
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, expected_graduation_date, id))
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ')
				PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)),
			PARTITION north_america
				VALUES IN ('US', 'CA') PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue))
		)

214:
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, expected_graduation_date, id))
	PARTITION BY LIST (country)
		(
			PARTITION australia
				VALUES IN ('AU', 'NZ') PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)),
			PARTITION north_america
				VALUES IN ('US', 'CA') PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue))
		)

230:
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, expected_graduation_date, id))
	PARTITION BY LIST (country)
		(
			PARTITION australia VALUES IN ('AU', 'NZ') PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)),
			PARTITION north_america
				VALUES IN ('US', 'CA') PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue))
		)

233:
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, expected_graduation_date, id))
	PARTITION BY LIST (country)
		(
			PARTITION australia VALUES IN ('AU', 'NZ') PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)),
			PARTITION north_america VALUES IN ('US', 'CA') PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue))
		)

450:
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, expected_graduation_date, id))
	PARTITION BY LIST (country)
		(PARTITION australia VALUES IN ('AU', 'NZ') PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)), PARTITION north_america VALUES IN ('US', 'CA') PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue)))

474:
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, expected_graduation_date, id))
	PARTITION BY LIST (country) (PARTITION australia VALUES IN ('AU', 'NZ') PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)), PARTITION north_america VALUES IN ('US', 'CA') PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue)))

632:
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CREATE TABLE students (id SERIAL8, name STRING, email STRING, country STRING, expected_graduation_date DATE, PRIMARY KEY (country, expected_graduation_date, id)) PARTITION BY LIST (country) (PARTITION australia VALUES IN ('AU', 'NZ') PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_au VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_au VALUES FROM ('2017-08-15') TO (maxvalue)), PARTITION north_america VALUES IN ('US', 'CA') PARTITION BY RANGE (expected_graduation_date) (PARTITION graduated_us VALUES FROM (minvalue) TO ('2017-08-15'), PARTITION current_us VALUES FROM ('2017-08-15') TO (maxvalue)))


